### DISCUSSION TITLE
Should the displayed value of an editable field be the edit with highest verification number, as oppose to the most recent edit?

### DISCUSSION THREAD

**Discussion by edwardchalstrey1** [tomalrussell/colouring-london#883] [Q&A]
... or the most commonly entered value, rather than the most recent

**Comment by tomalrussell** [comment]
This is not at all obvious to me - why should the displayed value not be the most recently edited?

What implications does this have for editing flows, from a user perspective:
- What happens when I edit an already-verified value? Does that new value show in any way?
- How do I see if a non-verified or less-verified value has been set?
- If one value is verified, how can I assert that another value should be the correct one?
- If an edit is malicious (however mildly so) and verified, how would I correct or remove it?
- How is "most commonly entered" calculated?
- How can I understand what values have been entered, how commonly, and how often verified?

And then in implementation on the backend:
- currently, all edits are recorded in the edit history, and the current value for each attribute is stored in the buildings table
- currently, all verifications (per user, building, attribute) are stored in another table; verification count for the current is calculated in the application code
- what data model changes would be required?
- what API changes would be required?

**Reply by edwardchalstrey1** [reply]
> What happens when I edit an already-verified value? Does that new value show in any way?
> How do I see if a non-verified or less-verified value has been set?

the idea so far would be you just have to look at the history

> If one value is verified, how can I assert that another value should be the correct one?
. If an edit is malicious (however mildly so) and verified, how would I correct or remove it?

These are important questions, since currently you'll only have the option to verify the displayed value, and cannot flag anything as malicious - perhaps we should have some mechanism for users flagging spam entries? More importantly, the system should probably highlight disputed entries - this might end up being quite a complex feature in its own right

> How is "most commonly entered" calculated?
> How can I understand what values have been entered, how commonly, and how often verified?

I was just thinking if there was some back and forth of some people entering one value and others changing it to another, the displayed value should be the one that was more common in the edit history - but it might make more sense to allow verification of past values, then display the **most verified**, but have a flag/warning/icon indicating when conflicting entries had been saved, giving the user the option to verify the one they thought was correct - what do you think?

**Reply by polly64** [reply]
would be good to chat about this on zoom at next meeting as such a complicated one
